### ISSUE TITLE
AoT compilation support

### ISSUE THREAD

**Issue by lzoubek** [pleerock/ng2-modal#21]
Hi, I am trying to compile my application using angular-cli and it's failing

```
Error: Unexpected value 'ModalModule' imported by the module 'SharedModule'
```

This is because `ng2-modal` was not built with `angular-cli` . I  found potential fix [here](https://github.com/dougludlow/ng2-bs3-modal/commit/6438b3236dbd8a2f28cf6b9d0f5c6a350415c05f)

Could you please incorporate it?

**Comment by pleerock** [comment]
can you provide more details? what exactly resolves the issue? The change you shown does seem to bring something that resolve the issue?

**Comment by lzoubek** [comment]
I think you need to build ng2-modal using `ngc` insteadof `tsc` and make sure you publish metadata.json file into npm repo. Here is the original issue [issue](https://github.com/dougludlow/ng2-bs3-modal/issues/105) in ng2-bs3-modal project

**Comment by marko220** [comment]
Hi - I am also facing this issue. Is this in the pipeline to be fixed?

**Comment by pleerock** [comment]
I'll take a look how to implement it once I get more free time. Will be great if someone can contribute here

**Comment by acassan** [comment]
Have you plan to fix it ? Same issue

**Comment by pleerock** [comment]
yes when I get more free time to investigate, would be great if someone create a PR for this

**Comment by Ninja101** [comment]
Can be closed \o/ #24

**Comment by lzoubek** [comment]
cool
